components/script/dom/htmlmediaelement.rs

```diff
@@ -1217,6 +1217,14 @@ impl HTMLMediaElement {
                 //    an unsupported format, or can otherwise not be rendered at all"
                 if self.ready_state.get() < ReadyState::HaveMetadata {
                     self.queue_dedicated_media_source_failure_steps();
+                    return;
+                }
+
+                // XXX(ferjm) We need to identify if the stream is seekable or
+                //            not to decide if we can loop. Ref. #22297
+                if self.Loop() {
+                    self.playback_position
+                        .set(self.default_playback_start_position.get());
```

Comment:
Instead of setting the `playback_position` here, we need to restart playback and seek to the `default_playback_start_position`.

Reply:
Thanks for the instruction, I try to work on it in coming few days.

p.s. I cannot compile it locally on my Mac because of the problem of -lstdc++ not found. I currently work on it on VM in cloud provider (turn on and off). Therefore the progress is quite slow. I am sorry about that.

Reply:
If you run `cargo update -p azure` I think the problem might go away.